hoomd/hpmc/update.py

```diff
@@ -15,387 +15,138 @@
 import hoomd
 
 
-class boxmc(_updater):
+class BoxMC(Updater):
     R""" Apply box updates to sample isobaric and related ensembles.
 
     Args:
 
-        mc (:py:mod:`hoomd.hpmc.integrate`): HPMC integrator object for system on which to apply box updates
-        betaP (:py:class:`float` or :py:mod:`hoomd.variant`): :math:`\frac{p}{k_{\mathrm{B}}T}`. (units of inverse area in 2D or
-                                                    inverse volume in 3D) Apply your chosen reduced pressure convention
-                                                    externally.
         seed (int): random number seed for MC box changes
+        betaP (float or :py:mod:`hoomd.variant`): :math:`\frac{p}{k_{\mathrm{B}}T}`. (units of inverse area in 2D or
+                                                    inverse volume in 3D) Apply your chosen reduced pressure convention externally.
+        trigger (hoomd.trigger.Trigger): Select the timesteps to perform box trial moves.
 
```

Comment:
Let's add a more extended paragaph length description of what BoxMC is between `Args` and `Attributes`. This should include a discussion of the weights.